.github/workflows/swift-toolchain.yml

```diff
@@ -3193,6 +3193,34 @@ jobs:
             Install-Package -Name WixToolset.Sdk -RequiredVersion 4.0.1 -Force
           }
 
+      - uses: actions/checkout@v4
+        if: matrix.arch == 'amd64'
+        with:
+          repository: microsoft/mimalloc
+          ref: refs/tags/v2.1.7
+          path: ${{ github.workspace }}/SourceCache/mimalloc
+          show-progress: false
+
+      - name: Build and apply mimalloc
+        if: matrix.arch == 'amd64'
```

Comment:
Should we be checking the OS as well as we are working towards the macOS toolchain?

Reply:
As the entire enclosing job is restricted to Windows (and the matrix doesn't have OSes yet), so this looks okay to me for now.
```
  package_tools:
    # TODO: Build this on macOS or make an equivalent Mac-only job
    if: inputs.build_os == 'Windows'
    ...
```